### PR TITLE
Add cargo-orthohelp: IR doc tooling, bridge pipeline, tests, and docs

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Add cargo-orthohelp: IR doc tooling with an ephemeral bridge pipeline to generate per-locale localised IR JSON from a target crate.
- Introduce a new cargo-orthohelp binary with modules for metadata discovery, bridge crate generation, caching, locale resolution, and per-locale IR emission.
- Include CLI, error handling, IR structures, and output writers, plus tests (unit and behavioural) and a fixture crate to exercise end-to-end workflows.
- Introduce an ExecPlan document for the bridge pipeline: `docs/execplans/4-1-1-ortho-config-help-bridge-pipeline.md` and update related docs/examples to reflect the new tooling.
- Update workspace layout to include the new crate and wire through docs/design and usage guides.
- Adds IR schema, localisation support, and bridge-based IR generation for per-locale output.

🌿 Generated by [Terry](https://www.terragonlabs.com)

---

ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/6b8f9830-60ac-46fb-b909-4a2013a8504d

## Changes
- Added new crate: `cargo-orthohelp` with the following modules/files:
  - Cargo.toml (new crate definition)
  - src/bridge.rs
  - src/cache.rs
  - src/cli.rs
  - src/error.rs
  - src/ir.rs
  - src/locale.rs
  - src/main.rs
  - src/metadata.rs
  - src/output.rs
  - src/schema.rs
- Added unit and integration tests for cargo-orthohelp:
  - cargo-orthohelp/tests/... (rstest-based tests, including fixture-based tests)
  - cargo-orthohelp/tests/rstest_bdd/... (behavioural tests and steps/scenarios)
- Added a fixture crate to exercise the tool flow: `tests/fixtures/orthohelp_fixture` with sample Cargo.toml, locales, and a minimal ORTHO config type.
- Updated workspace: include `cargo-orthohelp` in Cargo.toml workspace members.
- Documentation and design updates:
  - ExecPlan added: `docs/execplans/4-1-1-ortho-config-help-bridge-pipeline.md` (live plan for the bridge pipeline)
  - Updated usage/docs in `docs/cargo-orthohelp-design.md` and `docs/users-guide.md` to reflect the new tooling (per-locale IR emission, cache behavior, and no-build semantics)
  - Examples updated: `examples/hello_world` now includes Ort ho config metadata and usage guidance for cargo-orthohelp
- Codebase compatibility and IR changes:
  - Introduced IR structures and localisation helpers to support per-locale resolution and emission
  - Extended error types to cover bridge/build/execution/cache scenarios
  - Lightweight bridge-based approach to generate IR from a target crate without embedding tooling in the binary

## Rationale
- The new cargo-orthohelp tooling provides a formal, auditable bridge pipeline to produce per-locale IR JSON from a real crate, aligning with the design in `docs/cargo-orthohelp-design.md` and the roadmap. The ExecPlan documents the intended workflow while keeping a living-document approach for roadmap item 4.1.1. The approach avoids embedding behavior in the application and keeps the IR generation logic testable and reproducible via a small ephemeral bridge crate.

## Validation plan
- Build and run tests for the workspace to ensure compilation and test coverage pass (unit, integration, and behavioural tests).
- Exercise the bridge pipeline against the fixture crate (`orthohelp_fixture`) using:
  - --cache to reuse IR when valid
  - --no-build to bypass the bridge when cached IR is present
  - Per-locale output under a specified --out-dir (IR JSON files at <out>/ir/<locale>.json)
- Verify that per-locale IR JSON resolves Fluent IDs to translated text using embedded defaults plus consumer resources (locales/**). Validate fallback behavior when resources are missing or locales are unknown.
- Confirm the ExecPlan document is in place and describes constraints, progress, risks, and validation steps, and that it stays updated as work progresses.
- Ensure documentation, examples, and roadmap reflect the new flow and that tests cover the key end-to-end behaviours.

## Next steps
- Finalise CLI refinements and error messaging for edge-cases (e.g., missing cache when --no-build is requested).
- Expand tests to cover more locales and additional fixture variations.
- Iterate on the ExecPlan with any workflow decisions and changes observed during implementation.

---

🌿 Generated by [Terry](https://www.terragonlabs.com)

📎 **Task**: https://www.terragonlabs.com/task/5bda6320-de41-47a2-b00b-73dcdea0818c

## Summary by Sourcery

Introduce the cargo-orthohelp binary and bridge pipeline to generate per-locale localised IR JSON from ortho_config-enabled crates, and document the new workflow.

New Features:
- Add the cargo-orthohelp workspace crate with a CLI that discovers target packages, builds an ephemeral bridge, and emits per-locale localised IR JSON.
- Define an internal IR schema mirror and localisation layer to resolve Fluent IDs into concrete strings for each locale.
- Provide a fixture crate and user-facing guidance so consumers can configure ortho_config metadata and invoke cargo-orthohelp to generate IR.

Enhancements:
- Extend ortho_config IR structs to support deserialization for round-tripping JSON through the new tooling.
- Document the localised IR JSON schema, caching behaviour, and locale resource discovery in the design doc and users guide, and mark the bridge pipeline roadmap item as completed.
- Update the hello_world example with ortho_config metadata and instructions for generating documentation IR with cargo-orthohelp.

Build:
- Add the cargo-orthohelp crate and orthohelp_fixture test fixture to the workspace and bump rstest-bdd dependencies to v0.4.0.

Documentation:
- Add an ExecPlan documenting the cargo-orthohelp bridge pipeline and update design, roadmap, and user docs to describe the new IR generation flow and per-locale output.

Tests:
- Add unit tests for cache fingerprinting and locale resolution, plus rstest-bdd behavioural tests that exercise the full cargo-orthohelp bridge pipeline using the new fixture crate.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,8 +22,8 @@ keywords = ["config", "configuration", "cli", "env", "settings"]
 categories = ["config", "command-line-interface"]
 
 [workspace.dependencies]
-rstest-bdd = "0.3.2"
-rstest-bdd-macros = "0.3.2"
+rstest-bdd = "0.4.0"
+rstest-bdd-macros = "0.4.0"
```
[the remark this answers is not in the document]
Comment (leynos) [reply Cargo.toml]: @coderabbitai carets are unnecessary here as Cargo.toml provides implicit caret versioning
